### ISSUE TITLE
CLI: parsing steps output missing when run with --simple-output

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#1444]
### Description of Issue/Question

When running in a tty, the "parsing steps" stage leaves an hourglass when it completes, and the "Parsing . . . steps" message is duplicated:

```
Parsing orchestration ceph.stage.2 steps... ⏳

[parsing] on master
            |_ ceph.stage.2
               
[parsing] on *
            |_ ceph.monitoring.prometheus.exporters.node_exporter
               ceph.refresh
               
[parsing] on target158069064253.teuthology
            |_ ceph.admin.key
               ceph.mgr.key
               ceph.ganesha.key
               ceph.osd.key
               ceph.mon.key
               ceph.mds.key
               ceph.igw.key
               ceph.monitoring
               ceph.openattic.key
               ceph.rgw.key
               
Parsing ceph.stage.2 steps... ✓
```

Should the initial hourglass turn into a checkmark and omit the second, duplicate message? Or, alternatively, should the initial hourglass be omitted and also omit the second, duplicate message? Or the second duplicate message could be changed to "Parsing complete"?

Also, when run with `--simple-output`, the output is much less verbose:

```
Parsing orchestration ceph.stage.1 steps... done
```

Would it make sense for `--simple-output` mode to display the same information?

**Comment by gekios** [comment]
We hit the bug today as well while working with @swiftgist on PR CI test failures. Due to the CI running the stage with `--simple-output` the error wasn't in the output so we had to rerun the tests and manually run the stage to identify the issue. Below is the output difference with and without `--simple-output`

```target192168000116:/home/ubuntu # DEV_ENV=true timeout 60m deepsea --log-file=/var/log/salt/deepsea.log --log-level=debug salt-run state.orch ceph.stage.3 --simple-output
Starting stage: ceph.stage.3
Parsing stage ceph.stage.3 steps... fail

An error occurred when rendering one of the following states:
Traceback (most recent call last):
  File "/usr/bin/deepsea", line 11, in <module>
    load_entry_point('deepsea==0.8.13+2.g3f43be5b', 'console_scripts', 'deepsea')()
  File "/usr/lib/python2.7/site-packages/deepsea/deepsea.py", line 371, in main
    cli(prog_name='deepsea')
  File "/usr/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/deepsea/common.py", line 66, in func_wraper
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/deepsea/deepsea.py", line 356, in state_orch
    ret = run_stage(stage_name, hide_state_steps, hide_dynamic_steps, simple_output)
  File "/usr/lib/python2.7/site-packages/deepsea/stage_executor.py", line 79, in run_stage
    mon.parse_stage(stage_name)
  File "/usr/lib/python2.7/site-packages/deepsea/monitor.py", line 538, in parse_stage
    self._fire_event('stage_parsing_finished', None, None, ex)
  File "/usr/lib/python2.7/site-packages/deepsea/monitor.py", line 601, in _fire_event
    getattr(listener, event)(*args)
  File "/usr/lib/python2.7/site-packages/deepsea/monitors/terminal_outputter.py", line 45, in stage_parsing_finished
    for state in exception.states:
AttributeError: 'StateRenderingException' object has no attribute 'states'```

VS

```target192168000116:/home/ubuntu # DEV_ENV=true timeout 60m deepsea --log-file=/var/log/salt/deepsea.log --log-level=debug salt-run state.orch ceph.stage.3 
Starting stage: ceph.stage.3
Parsing ceph.stage.3 steps... ⏳

[parsing] on master
            |_ ceph.stage.3
               
[parsing] on I@roles:mgr and I@cluster:ceph
            |_ ceph.mgr
               ceph.mgr.keyring
               ceph.tuned.mgr
               
[parsing] on I@roles:mgr
            |_ ceph.processes.mgr
               
[parsing] on *
            |_ ceph.osd.grains
               
[parsing] on I@roles:storage and I@cluster:ceph
            |_ ceph.tuned.osd
               ceph.sysctl
               ceph.osd
               
[parsing] on I@roles:mon and I@cluster:ceph
            |_ ceph.tuned.mon
               ceph.mon
               
[parsing] on I@roles:mon
            |_ ceph.processes.mon
               
[parsing] on target192168000116.teuthology
            |_ ceph.osd.auth
               ceph.osd.restart
               ceph.monitoring.prometheus.exporters.ceph_exporter
               ceph.configuration.check
               ceph.mgr.auth
               ceph.configuration.create
               ceph.wait
               ceph.mon.restart
               ceph.wait.until.all_osds_in
               ceph.monitoring.prometheus.exporters.rbd_exporter
               ceph.mgr.restart
               ceph.pool
               
Parsing ceph.stage.3 steps... ❌

An error occurred when rendering one of the following states:
    - ceph.wait.until.all_osds_in (/srv/salt/ceph/wait/until/all_osds_in)

Error description:
  - No matching sls found for 'ceph.wait.until.all_osds_in' in env 'base'```

**Comment by smithfarm** [comment]
Raising priority because the bug negatively affects developers' ability to understand bugs flagged by the CI. (The CI always passes `--simple-output` to the DeepSea CLI to avoid getting escape sequences in the console log.)

**Comment by rjfd** [comment]
@smithfarm this bug was fixed in master in PR #1571 . Probably we are missing a backport somewhere, which version of SES is triggering this bug?

**Comment by smithfarm** [comment]
Fixed by https://github.com/SUSE/DeepSea/pull/1781
